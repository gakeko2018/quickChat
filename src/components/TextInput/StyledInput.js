import React from "react";
import {
  StyleSheet,
  Text,
  TextInput,
  KeyboardAvoidingView,
  TouchableOpacity
} from "react-native";
import { colors, margin } from "../../styles/base";

const StyledInput = ({ style, placeholder, onChangeText, value }) => {
  return (
    <TextInput
      style={[styles.input, style]}
      placeholder={placeholder}
      onChangeText={onChangeText}
      value={value}
    />
  );
};

const styles = StyleSheet.create({
  input: {
    fontSize: 20,
    textAlign: "center",
    margin: margin.sm,
    borderRadius: 10,
    backgroundColor: colors.primaryBackground,
    color: colors.secondaryFontColor,
    minHeight: 30,
    borderWidth: 1,
    borderColor: colors.tertiaryBackground,
    borderStyle: "solid"
  }
});

export default StyledInput;
