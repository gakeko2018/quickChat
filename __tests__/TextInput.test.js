import React from "react";
import { shallow } from "enzyme";
import StyledInput from "../src/components/TextInput/StyledInput";

import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <StyledInput
      placeholder="Message"
      onChangeText={() => {}}
      value={"New Message"}
    />
  );
});

describe("TextInputs renders correctly", () => {
  it(" StyledInput renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
