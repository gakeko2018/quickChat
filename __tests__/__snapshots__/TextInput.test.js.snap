// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputs renders correctly  StyledInput renders correctly 1`] = `ShallowWrapper {}`;
